integration-libs/cdc/user-profile/update-email/cdc-update-email.module.ts

```diff
@@ -6,23 +6,41 @@
  */
 
 import { CommonModule } from '@angular/common';
-import { NgModule } from '@angular/core';
+import { NgModule, inject } from '@angular/core';
 import { FormsModule, ReactiveFormsModule } from '@angular/forms';
 import { RouterModule } from '@angular/router';
 import {
   CmsConfig,
   I18nModule,
   provideDefaultConfig,
+  provideDefaultConfigFactory,
   UrlModule,
 } from '@spartacus/core';
 import {
   FormErrorsModule,
   PasswordVisibilityToggleModule,
   SpinnerModule,
 } from '@spartacus/storefront';
-import { UpdateEmailComponentService } from '@spartacus/user/profile/components';
+import {
+  MyAccountV2EmailComponentService,
+  USE_MY_ACCOUNT_V2_EMAIL,
+  UpdateEmailComponentService,
+} from '@spartacus/user/profile/components';
 import { CDCUpdateEmailComponentService } from './cdc-update-email-component.service';
 
+const myAccountV2EmailWithCDCMapping: CmsConfig = {
+  cmsComponents: {
+    UpdateEmailComponent: {
+      providers: [
+        {
+          provide: MyAccountV2EmailComponentService,
```

Comment:
when `USE_MY_ACCOUNT_V2_EMAIL` is true, provide `MyAccountV2EmailComponentService` instead of `UpdateEmailComponentService`